Restructure - Populate learn (#1080)
iotaledger/iota-wiki: docs/learn/chrysalis/core-concepts/switch-to-UTXO.md

---
description: Learn about the switch from account balance models to Unspent Transaction Output (UTXO) in IOTA.
keywords: [ "IOTA", "UTXO", "Balance Models", "Conflicts", "Reattachments", "Colored Coins" ]
---

# Switch to UTXO

UTXO means "unspent transaction output". 
This means that instead of tracking the balances of addresses, you also keep track of where balances came from and 
where they go when they are spent.

## The Problem with the Balance Mode

Before Chrysalis, IOTA used an account balance model to track addresses.
This meant every address had a single value associated with it, its current balance.

> This meant the ledger tracked **Address1**, which was associated with **Balance1**, **Address2** was associated with *
> *Balance2**, and so on.

Reviewer: ```suggestion
This meant the ledger tracked **Address1**, which was associated with **Balance1**, **Address2** was associated with **Balance2**, and so on.
```

### Double-spends

The main problem with this model was dealing with double-spends, also known as conflicts.
It took a lot of work to determine which of multiple transactions performed a double spend and which used legitimate
funds.

> This was not a problem in the previous protocol since the “heaviest sub-tangle wins” rule only had to ensure that the
> addresses of a particular sub-tangle never turned negative.
> However, Chrysalis introduced a voting-based Coordicide solution.
> This meant that it was necessary to identify conflicts quickly and precisely so we could reduce the amount of voting
> required to resolve these conflicts.

Reviewer: ```suggestion
As with IOTA 2.0 a voting-based Coordicide solution was introduced, this meant that it was necessary to identify conflicts quickly and precisely so we could reduce the amount of voting required to resolve these conflicts.
```

### Reatachments

Another problem with using an account balance model is reattachments.
If somebody ever receives funds at an address already used to send tokens, anybody can just reattach the previous
transaction and empty the address again,
even if they do not control the private key.
There have been many cases in DLTs where this was used as an “attack vector” when users didn’t follow the advice to use

> addresses only once.

Reviewer: As this is special to how IOTA did it and not in general to account models, I would remove this section. We could also replace it with a link to https://wiki.iota.org/tips/tips/TIP-0017/

> ### Reatachments
> 
> Another problem with using an account balance model is reattachments.
> If somebody ever receives funds at an address already used to send tokens, anybody can just reattach the previous
> transaction and empty the address again,
> even if they do not control the private key.
> There have been many cases in DLTs where this was used as an “attack vector” when users didn’t follow the advice to use
> addresses only once.

Author: ```suggestion
```

## The UTXO Solution

Since Chrysalis, IOTA switched to the UTXO model to keep track of balances.
Every address contains its total balance and multiple sub-balances with a marker indicating precisely which transaction
created the funds.
Every coin on an address can be uniquely identifiable, and every transaction names the exact coins it wants to move.
This switch significantly improved how we identify conflicts and, therefore, also stops malicious actors from spending
newly received funds by reattaching an old transaction.

### Support for Ed25519

An additional benefit is that the UTXO model supports [Ed25519 addresses](EdDSA-support.md),
meaning that addresses in Chrysalis are reusable and can be safely used to send funds more than once.

> :::tip Learn More
> 
> You can learn more about the [switch to UTXO](https://wiki.iota.org/tips/tips/TIP-0007) in
> the [TIPS section](../tips.md).
> 
> :::

Reviewer: ```suggestion
:::tip Learn More

You can learn more about the [switch to UTXO](https://wiki.iota.org/tips/tips/TIP-0007) and [Ed25519](https://wiki.iota.org/tips/tips/TIP-0017/) in
the [TIPS section](../tips.md).

:::
```

> :::

Author: ```suggestion
:::tip Learn More
	
You can learn more about the [switch to UTXO](https://wiki.iota.org/tips/tips/TIP-0007) and [Ed25519](https://wiki.iota.org/tips/tips/TIP-0017/) in
the [TIPS section](../tips.md).

:::
```

---
description: Learn about the switch from account balance models to Unspent Transaction Output (UTXO) in IOTA.
keywords:
  [
    'IOTA',
    'UTXO',
    'Balance Models',
    'Conflicts',
    'Reattachments',
    'Colored Coins',
  ]
---

# Switch to UTXO

UTXO means "unspent transaction output".
This means that instead of tracking the balances of addresses, you also keep track of where balances came from and
where they go when they are spent.

## The Problem with the Balance Mode

> Before Chrysalis, IOTA used an account balance model to track addresses.

Reviewer 2: ```suggestion
Before Chrysalis, IOTA used an account balance model to track balances.
```

---
description: Learn about the switch from account balance models to Unspent Transaction Output (UTXO) in IOTA.
keywords:
  [
    'IOTA',
    'UTXO',
    'Balance Models',
    'Conflicts',
    'Reattachments',
    'Colored Coins',
  ]
---

# Switch to UTXO

UTXO means "unspent transaction output".
This means that instead of tracking the balances of addresses, you also keep track of where balances came from and
where they go when they are spent.

## The Problem with the Balance Mode

Before Chrysalis, IOTA used an account balance model to track addresses.
This meant every address had a single value associated with it, its current balance.

> This meant the ledger tracked **Address1**, which was associated with **Balance1**, **Address2** was associated with \*
> \*Balance2\*\*, and so on.

Reviewer: ```suggestion
This meant the ledger tracked **Address1**, which was associated with **Balance1**, **Address2** was associated with **Balance2**, and so on.
```

### Double-spends

The main problem with this model was dealing with double-spends, also known as conflicts.

> It took a lot of work to determine which of multiple transactions performed a double spend and which used legitimate
> funds.

Reviewer: ```suggestion
It took a lot of work to determine which of multiple transactions performed a double spend and which used legitimate funds.
```

> ### Reatachments
> 
> Another problem with using an account balance model is reattachments.
> If somebody ever receives funds at an address already used to send tokens, anybody can just reattach the previous
> transaction and empty the address again,
> even if they do not control the private key.
> 

Reviewer 2: ```suggestion
### Reattachments

Another problem with using an account balance model is reattachments.
If somebody ever receives funds at an address already used to send tokens, anybody can just reattach the previous
transaction and empty the address again,
even if they do not control the private key.
```

or even better

```suggestion
```
:trollface:

Reviewer 2: Because this has nothing to do with Account vs UTXO

## The UTXO Solution

Since Chrysalis, IOTA switched to the UTXO model to keep track of balances.

> Every address contains its total balance and multiple sub-balances with a marker indicating precisely which transaction
> created the funds.
> Every coin on an address can be uniquely identifiable, and every transaction names the exact coins it wants to move.

Reviewer: ```suggestion
```

I would remove this. It is confusing at best and coins if kept should be replaces with Native tokens. But still weird

---
description: This article provides an in-depth look into IOTA's shift from WOTS addresses to the UTXO model and the motivations and processes behind this migration.
keywords: [ 'IOTA', 'WOTS', 'UTXO', 'Migration', 'Chrysalis', 'Ed25519', 'Security', 'Protocol Design', 'User Experience' ]
---
# Switch to UTXO 

> With Chrysalis, IOTA moved away from the Winternitz One-Time Signature (WOTS) system to the Unspent Transaction Output (

Reviewer: Isn't this still a little bit weird 😅

I mean we switched from account based to UTXO
And from WOTS to Ed25519 addresses.

UTXO) model, addressing several issues related to security, protocol design, and user experience.

## Challenges with the Account Balance Model

Before the transition to Chrysalis, IOTA operated on an account model.

> \Every address held a distinct balance value.

Reviewer: Why can't I make a suggestion? 😢 
Every address held a distinct balance value.

Author: ```suggestion
Every address held a distinct balance value.
```

However, when double-spends or conflicts arose, discerning legitimate transactions from double-spends became complex.
This model posed two main challenges:

- **Conflict Handling**: Identifying the transaction performing a double-spend was intricate, leading to extensive
  "conflict sets" and reducing the efficiency in addressing conflicts.

> - **Reattachments**: In cases where funds were received at an already spent address, it became possible for anyone to

Reviewer: I'm actually not sure if that is true? I thought the problem was somehow exposing your private key with every signature of an already spend address or something like that 🤔.
